layers/ip6.go

```diff
@@ -34,6 +75,7 @@ type IPv6 struct {
 	HopLimit     uint8
 	SrcIP        net.IP
 	DstIP        net.IP
+	RoutingDstIP net.IP
```

Comment:
Cool, thanks!  Happy to accept any suggestions you might have.

I think it should be fine for IPv6 to look at its own options... and it
seems easier than having the options have to go back and find the IPv6
layer, which we currently can't do without changing the packet builder.

On Fri, Sep 11, 2015 at 10:58 AM, randstr <notifications@github.com> wrote:

> In layers/ip6.go
> <https://github.com/google/gopacket/pull/120#discussion_r39298121>:
>
> > @@ -34,6 +75,7 @@ type IPv6 struct {
> >  	HopLimit     uint8
> >  	SrcIP        net.IP
> >  	DstIP        net.IP
> > +	RoutingDstIP net.IP
>
> It'd have to return the final destination of the routing header, which is
> not a concern at all once the header is found in the payload.
>
> Having one layer parse the others doesn't feel quite right yet but maybe
> it'll work. I'll give it some more thought.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/gopacket/pull/120/files#r39298121>.
>